crates/sui-json-rpc-tests/tests/name_service_tests.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+use sui_json_rpc::name_service;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress},
+    collection_types::VecMap,
+};
+
+#[test]
+fn test_parent_extraction() {
+    let mut name = name_service::Domain::from_str("leaf.node.test.sui").unwrap();
+
+    assert_eq!(name.parent().to_string(), "node.test.sui");
+
+    name = name_service::Domain::from_str("node.test.sui").unwrap();
+
+    assert_eq!(name.parent().to_string(), "test.sui");
+}
+
+#[test]
+fn test_expirations() {
+    let system_time: u64 = 100;
+
+    let mut name = name_service::NameRecord {
+        nft_id: sui_types::id::ID::new(ObjectID::random()),
+        data: VecMap { contents: vec![] },
+        target_address: Some(SuiAddress::random_for_testing_only()),
+        expiration_timestamp_ms: system_time + 10,
+    };
```

Comment:
shouldn't we instead determine expiration from ... say the Clock object or something? Not sure, just asking qs

Reply:
It's a simplification on the test per @amnn comment! We'll use the name_service for all different systems (indexer, indexer_v2, graphql), and each one will supply the expiration on its own!